EIPS/eip-2020.md

```diff
@@ -0,0 +1,268 @@
+---
+eip: 2020
+title: Ethereum Provider API
+author: Ryan Ghods (@ryanio), Marc Garreau (@marcgarreau)
+discussions-to: https://ethereum-magicians.org/t/eip-2020-ethereum-provider/640
+status: Draft
+type: Standards Track
+category: Interface
+created: 2018-06-30
+requires: 1102
+---
+
+## Summary
+
+This proposal formalizes an Ethereum Provider API.
+
+## API
+
+### Send
+
+```js
+ethereum.send(method: String, params: Array<any>): Promise<any>;
+```
+
+Promise resolves with `result` or rejects with `Error`.
+
+See [methods](https://github.com/ethereum/wiki/wiki/JSON-RPC#json-rpc-methods).
+
+### Subscriptions
+
+#### Subscribe
+
+```js
+ethereum.subscribe(subscriptionType: String, params: Array<any>): Promise<any>;
+```
+
+Promise resolves with `subscriptionId: String` or rejects with `Error`.
+
+See [subscription types](https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB#supported-subscriptions).
+
+Results emit on `subscriptionId` using [EventEmitter](https://nodejs.org/api/events.html). Attach listeners with:
+
+```js
+ethereum.on(subscriptionId, listener: (result: any) => void): this;
+```
+
+The event emits with `result`, the subscription result or an `Error` object.
+
+#### Unsubscribe
+
+```js
+ethereum.unsubscribe(subscriptionId: String): Promise<any>;
+```
+
```

Comment:
I would add the following here "Promise resolves with `subscriptionId: String` or rejects with `Error`."

Maybe it would be better to write it like this: 
```ethereum.unsubscribe(subscriptionId: String): Promise<String|Error>;``` 
instead of: 
```ethereum.unsubscribe(subscriptionId: String): Promise<any>;```

Reply:
Great idea thanks, will amend.